packages/template-retail-react-app/app/components/bonus-product-modal/index.test.jsx

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2025, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React from 'react'
+import PropTypes from 'prop-types'
+import {render, screen, fireEvent} from '@testing-library/react'
+import {BonusProductModal} from '@salesforce/retail-react-app/app/components/bonus-product-modal'
+import {useBonusProductModalContext} from '@salesforce/retail-react-app/app/hooks/use-bonus-product-modal'
+import {useProducts} from '@salesforce/commerce-sdk-react'
+
+// Mock the hooks
+jest.mock('@salesforce/retail-react-app/app/hooks/use-bonus-product-modal')
+jest.mock('@salesforce/commerce-sdk-react')
+
+// Mock provider component
+const MockProvider = ({children}) => {
+    return <div data-testid="mock-provider">{children}</div>
+}
+
+MockProvider.propTypes = {
+    children: PropTypes.node.isRequired
+}
+
+describe('BonusProductModal', () => {
+    const mockContextValue = {
+        isOpen: true,
+        onClose: jest.fn(),
+        data: {
+            newBonusItems: [
+                {
+                    bonusProducts: [
+                        {
+                            id: '1',
+                            productId: '1',
+                            productName: 'Product 1',
+                            title: 'Product 1'
+                        },
+                        {
+                            id: '2',
+                            productId: '2',
+                            productName: 'Product 2',
+                            title: 'Product 2'
+                        }
+                    ],
+                    maxBonusItems: 2
+                }
+            ]
+        }
+    }
+
+    const mockProductData = {
+        data: [
+            {
+                imageGroups: [
+                    {
+                        viewType: 'small',
+                        images: [{link: 'test-image.jpg'}]
+                    }
+                ]
+            }
+        ]
+    }
+
+    beforeEach(() => {
+        useBonusProductModalContext.mockReturnValue(mockContextValue)
+        useProducts.mockReturnValue({
+            data: mockProductData,
+            isLoading: false
+        })
+    })
+
+    afterEach(() => {
+        jest.clearAllMocks()
+    })
+
+    it('renders modal when isOpen is true', () => {
+        render(
```

Comment:
Nit: Let's use `renderWithProvider` to have a more accurate representation of what the DOM would be like for this component